### PR TITLE
Pass config to cadence matching and frontend services during initialization

### PR DESCRIPTION
**Pull request by tamer-eldeeb**
Replace constant config with a config struct that can be passed to the service during initialization.
Issue #286

### CODE REVIEW

**Comment by tamer-eldeeb** [comment]
History will be done in a separate PR as it will need more serious refactoring (#109) first

File: service/matching/matchingEngine.go [hunk 1]
```diff
@@ -63,20 +61,15 @@ type taskListID struct {
...
-
 var (
 	// EmptyPollForDecisionTaskResponse is the response when there are no decision tasks to hand out
 	emptyPollForDecisionTaskResponse = m.NewPollForDecisionTaskResponse()
 	// EmptyPollForActivityTaskResponse is the response when there are no activity tasks to hand out
 	emptyPollForActivityTaskResponse   = workflow.NewPollForActivityTaskResponse()
 	persistenceOperationRetryPolicy    = common.CreatePersistanceRetryPolicy()
 	historyServiceOperationRetryPolicy = common.CreateHistoryServiceRetryPolicy()
-	emptyGetTasksRetryPolicy           = createEmptyGetTasksRetryPolicy()
+	//emptyGetTasksRetryPolicy           = createEmptyGetTasksRetryPolicy()
```
Comment (venkat1109) [inline service/matching/matchingEngine.go]: remove commented code

File: service/matching/service.go [hunk 2]
```diff
@@ -21,21 +21,59 @@
...
+}
+
+// NewConfig returns new service config with default values
+func NewConfig() *Config {
+	return &Config{
+		EnableSyncMatch:              true,
+		LongPollExpirationInterval:   time.Minute,
+		EmptyGetRetryInitialInterval: 100 * time.Millisecond,
+		EmptyGetRetryMaxInterval:     1 * time.Second,
+
```
Comment (venkat1109) [inline service/matching/service.go]: delete empty line

File: service/matching/service.go [hunk 3]
```diff
@@ -21,21 +21,59 @@
...
+		RangeSize:         100000,
+		GetTasksBatchSize: 1000,
+		UpdateAckInterval: 10 * time.Second,
+
```
Comment (venkat1109) [inline service/matching/service.go]: delete empty line

File: service/matching/handler.go [hunk 4]
```diff
@@ -40,16 +40,18 @@ var _ m.TChanMatchingService = (*Handler)(nil)
...
 	engine          Engine
+	config          *Config
 	metricsClient   metrics.Client
 	startWG         sync.WaitGroup
 	service.Service
 }
 
 // NewHandler creates a thrift handler for the history service
-func NewHandler(taskPersistence persistence.TaskManager, sVice service.Service) (*Handler, []thrift.TChanServer) {
+func NewHandler(taskPersistence persistence.TaskManager, sVice service.Service, config *Config) (*Handler, []thrift.TChanServer) {
```
Comment (venkat1109) [inline service/matching/handler.go]: nit: can we make the order of args to all handlers somewhat consistent.. i.e. can we make *Config always the first arg to NewHandler() for all services ?

Comment (tamer-eldeeb) [inline service/matching/handler.go]: The "convention" I followed is to have config after the service.Service object. So how about this convention:
NewHandler(service.Service, config, extra objects needed)

Comment (venkat1109) [inline service/matching/handler.go]: sounds good !

**Pull request merged**